Open SSL v 1.1 support #204

iddqd13 opened the pull request:
Hello!
I tried to build libsourcey with OpenSSL and got errors. Also saw https://github.com/sourcey/libsourcey/issues/204 Issue.
i changed some files according to https://github.com/pocoproject/poco/commit/bfaa161c61cca06526eaf238bfdddd190381616b and OpenSSL Documentation.
Please, Check this Pull Request, maybe it will be useful.
i checked local build - build successfull ( OpenSSL 1.1.0 -stable , gcc 6.3. , Debian Stretch )

P.S. Sorry for my English. I'm learning it yet. Thanks.

auscaster committed:
Merge pull request #182 from artemiuzzz/master

fixed dereferencing null pointer

auscaster committed:
Fix #197 applying empty pixel format

auscaster committed:
Enable webrtcrecorder and webrtccapturer samples

Commit:
Fixed websocket handshaking issue with latest Firefox

auscaster committed:
Merge pull request #205 from lutsykigor/master

Fixed websocket handshake issue with latest Firefox

Commit:
Added OpenCV and WebRTC integration sample app

lutsykigor committed:
Removed local config

lutsykigor committed:
Added comments for Chrome support

lutsykigor committed:
Added comments regarding Chrome and Firefox usage

auscaster committed:
Merge pull request #206 from lutsykigor/webrtc-opencv-demo

Added OpenCV and WebRTC integration sample app

auscaster committed:
Add HTTPS echo server sample

auscaster committed:
Merge branch 'master' of github.com:sourcey/libsourcey

auscaster committed:
Only include opencvanalyzer sample if HAVE_OPENCV is set

auscaster committed:
Add missing i420_buffer.h include

cksmith committed:
Add .gitattributes file to deine line ending behaviour

cksmith committed:
Fix line endings (CRLF->LF)

auscaster committed:
Merge pull request #223 from cksmith/fix-line-endings

Fix line endings

R-Gerard committed:
Fix logic error when calling _encoder.init()

auscaster committed:
Merge pull request #230 from R-Gerard/streamrecorder_init

Fix logic error when calling _encoder.init()

auscaster committed:
Update WebRTC impl to support latest branch_heads/66

auscaster committed:
Merge branch 'master' of github.com:sourcey/libsourcey

Conflicts:
	cmake/FindWebRTC.cmake
	src/webrtc/include/scy/webrtc/audiopacketmodule.h
	src/webrtc/include/scy/webrtc/multiplexmediacapturer.h
	src/webrtc/include/scy/webrtc/peer.h
	src/webrtc/include/scy/webrtc/peermanager.h
	src/webrtc/include/scy/webrtc/streamrecorder.h
	src/webrtc/include/scy/webrtc/util.h
	src/webrtc/include/scy/webrtc/videopacketsource.h
	src/webrtc/src/audiopacketmodule.cpp
	src/webrtc/src/multiplexmediacapturer.cpp
	src/webrtc/src/streamrecorder.cpp

auscaster committed:
Resolve merge conflicts

auscaster committed:
Update WebRTC dependency in Dockerfile
Fix webrtccapturer sample

auscaster committed:
Release 1.1.2

auscaster committed:
Update StreamRecorder to support latest WebRTC

auscaster committed:
Release 1.1.3

auscaster committed:
Fix #171 HTTP form progress updates

auscaster committed:
Add rate limiting and realtime PTS calculation to media capture
Fix mediaserver demo

auscaster committed:
Merge branch 'master' of github.com:sourcey/libsourcey

auscaster committed:
Release 1.1.4

iddqd13 committed:
Open SSL v 1.1 support #204

iddqd13 committed:
Update x509certificate.cpp

Remove extra const_cast

iddqd13 committed:
Update hash.cpp

using internal::api(EVP_MD_CTX_cleanup(_ctxPtr));  instead of internal::api(EVP_MD_CTX_reset(_ctxPtr));

infinityproject16 committed:
fixed c++ build error narrowing conversion of '-1' from 'int' to 'char' on armv7

auscaster committed:
Merge pull request #265 from blackforest-tom/master

Fixed C++ build error "narrowing conversion"

infinityproject16 committed:
Fixed constant names for newer versions of ffmpeg

infinityproject16 committed:
Removed empty line (code style)

mr-kakorin committed:
Fixed usage of old-version webrtc::VideoFrame

In newer versions(including webrtc-22215-ab42706) of webrtc methods DataY (and DataU, DataV) moved from VideoFrameBuffer to PlanarYuv8Buffer which is implemented by I420Buffer. To access instance of I420Buffer GetI420() should be called.

auscaster committed:
Merge pull request #271 from inventive-seven/opencv-example-fix

Fixed usage of old-version webrtc::VideoFrame

auscaster committed:
Merge pull request #267 from blackforest-tom/master

Support newer ffmpeg versions

tuduongquyet commented on src/crypto/src/hash.cpp:
```diff
@@ -49,15 +57,26 @@ void Hash::reset()
     internal::api(EVP_MD_CTX_cleanup(&_ctx));
     internal::api(EVP_DigestInit(&_ctx, _md));
+#else
+    internal::api(EVP_MD_CTX_cleanup(_ctxPtr));
```
```suggestion
        EVP_MD_CTX_free(_ctxPtr);
```

For OpenSSL 1.1.1

kapulkin committed:
[*] code updates to fit new webrtc (not final)

kapulkin committed:
[*] webrtc compile definitions for MacOS

kapulkin committed:
[*] excluding code, not compatible with modern WebRTC. Outdated code fix.

Commit:
Fix fedora33 compiration

use of opaque EVP_MD_CTX
Add macro switch for openssl version

auscaster committed:
Merge pull request #289 from lighta/feat/openssl_1.1+_support

Fix fedora33 compilation

auscaster committed:
Merge pull request #285 from kapulkin/newWebrtc

New webrtc

auscaster committed:
Merge pull request #203 from sicxs/master

change receivers to share_ptr

auscaster committed:
feat!: C++20 modernisation

build: cmake 3.21+, target-based deps, FetchContent, CMakePresets
build: github actions CI (linux/mac/win, sanitizers, coverage)
build: docker ubuntu 24.04, vcpkg port, cpack

deps: libuv 1.50, openssl 3.x, ffmpeg 6+, llhttp 9.2.1
deps: nlohmann/json 3.11.3, zlib 1.3.1, opencv 4.x find module
deps: removed webrtc, poco, rtaudio, dshow, vendored sources

lang: c++20 minimum, pragma once, enum class, using aliases
lang: smart pointers throughout, noexcept destructors, nodiscard
lang: string_view, structured bindings, constexpr, shared_mutex
lang: std::filesystem, std::chrono, zero raw new/delete in core

security: openssl 3.x migration, tls 1.2 minimum enforced
security: alpn, sni, hostname verification (SSL_set1_host)
security: pacm sha256 checksums, ssl verification, path traversal protection

fix: all assert() replaced with runtime checks (~110 across 30 files)
fix: stream shutdown eof handled as graceful close
fix: websocket rfc 6455 compliance (rsv, opcode, mask, ping/pong/close)
fix: websocket fragmentation and partial frame buffering
fix: ffmpeg 6.x const oformat, missing avcodec include
fix: mediacapture loop seek bug, multiplexencoder thread safety
fix: taskrunner unique_ptr ownership, scheduler memory leaks
fix: stream write backpressure via high water mark
fix: FindFFmpeg.cmake rewritten for pkg-config

test: comprehensive coverage across all 15 modules
test: video encoder/decoder, audio decoder standalone tests
test: http round-trip, websocket frame encoding, pacm json tests
test: timer, ipc, logger, packetstream overflow tests

docs: readme, changelog, build.md, pacm/pluga readmes updated

auscaster committed:
chore: gitignore test output artifacts

auscaster committed:
chore: update pacm and pluga submodule refs

auscaster committed:
refactor: use fs:: wrappers instead of raw std::filesystem

auscaster committed:
chore: update pacm submodule ref

auscaster committed:
chore: update pacm and pluga submodule refs

auscaster committed:
chore: update pacm and pluga submodule refs

auscaster committed:
Merge branch 'master' into openssl_11_support
